docs/archive-data.md

```diff
@@ -0,0 +1,314 @@
+      URI: "file:///tmp/temporal_archival/development"
+```
+
+In the configuration above we see that:
+
+1. Archival is enabled at the cluster level.
+2. Namespaces can use either the local filestore provider or the Google Cloud provider.
+3. New Namespaces will default to the local provider, 
+
+:::tip
+
+**Want to disable Archival?**
+
+You can disable Archival by setting `archival.history.state` and `namespaceDefaults.archival.history.state` to `"disabled"`.
+
+Example:
+
+```yaml
+archival:
+  history:
+    state: "disabled"
+
+namespaceDefaults:
+  archival:
+    history:
+      state: "disabled"
+```
+
+:::
+
+### Namespace creation
+
+The Archival URI is set when a Namespace is created. If you do not specify the URI when creating a Namespace, it will use the value of `defaultNamespace.archival.history.URI` from the `config/development.yaml` file. Once set, the URI can not be changed. Each Namespace supports a single Archival URI, but each Namespace can use a different URI (archiver).
+
+At the cluster level, `archival.history.state` must be set to `"enabled"` for Archival to work at the Namespace level for Even Histories. Once enabled at the cluster level, a Namespace can safely switch between `enabled` and `disabled` states.
+
+Archival is supported in global Namespaces (Namespaces spanning multiple clusters). When Archival is running in a global Namepsace, it will first run on the active cluster and then some time later it will run on the standby cluster. Before archiving, a history check is done to see what has been previously uploaded.
+
+## Archive reads
+ 
+To retrieve archived Event Histories you will need the Workflow Id and the Run Id:
+
+```bash
+$ ./temporal --ns <namespace> wf show --wid <workflowId> --rid <runId>
+```
+
+:::tip
+
+Want to test this feature locally? Start by running a Temporal server:
+
+```bash
+$ ./temporal-server start
+```
+
+Then register a new Namespace with Archival enabled. 
+
+```bash
+$ ./tctl --ns samples-namespace namespace register --gd false --history_archival_status enabled --retention 0
+```
+
+Next, run a sample workflow such as the [helloworld temporal sample](https://github.com/temporalio/temporal-go-samples/tree/master/helloworld).
+
+Once the sample execution has finished, you can view the archived Event Histories by copying the `workflowId` and `runId` of the completed Workflow from the log output and running:
+
+```bash
+$ ./temporal --ns samples-namespace wf show --wid <workflowId> --rid <runId>
+```
+
+:::
+
+## Custom archiver
+
+Temporal does not intend to limit you to specific archive locations. To use a provider that is not currently supported you may create your own archiver.
+
+### Create a new package
+
+The first step is to create a new package for your implementation in [`/common/archiver`](https://github.com/temporalio/temporal/tree/master/common/archiver). Create a new directory in the archiver folder and arrange the  structure to look like the following:
+
+```
```

Comment:
The way this is styled and presented is sort of weird.

Why us "./common/archiver" using a relative path for example